doc/adrs/070_add_the_possibility_to_hide_or_fade_a_graphic_element.adoc

```diff
@@ -0,0 +1,67 @@
+= ADR-070 - Add the possibility to hide or fade a graphic element
+
+== Context 
+
+Sirius Web displays all elements of a diagram but in large diagram, the user may want to focus only on certain elements that are more relevant to its use.
+
+== Decision
+
+We will add to diagram elements two new states: faded and hidden. The first one keep the element visible but more attenuated (transparency or shades of gray). The second one totally hide the graphic element and the layout ignore it. In this case, the representation will propose a way to open a dialog displaying hidden elements with the possibility to make them visible again.
+
+=== Backend
+
+==== Auto-layout
+
+In Sirius Web, the auto-layout is managed by ELK. Sirius gives to ELK the graph and get from ELK a layouted graph. This is this layouted graph that is saved in database. When an element is hidden, we need to keep it in memory to allow the user to display it again. 
+
+The first solution is to totally remove a hidden element from the saved graph and to store it separately. But the element is just hidden, not removed and if a part of the application iterates over all elements of the graph, this part can not access to hidden elements. 
+
+The second solution is to persist these hidden elements after the auto-layout was applied. A way to do that is to indicate to ELK to not consider the element in the layout computation. For this, ELK proposes the https://www.eclipse.org/elk/reference/options/org-eclipse-elk-noLayout.html[`noLayout`] property applies to nodes, border nodes, edges and labels. With this solution hidden elements are still present in the diagram, to invisibilize them we can use CSS.
```

Comment:
Just in case: According to the documentation, the `noLayout` option does not say that such elements will be hidden. They will still be visible in the frontend but with unknown size and position.

Reply:
So we may need to ignore on the frontend those elements and the edges related to them

Reply:
Maybe my text isn't very explicit: I propose to use CSS properties to hide elements such as edges or nodes. Indeed, the `noLayout` is just to avoid the "presence" of these hidden elements on the layout of the diagram.
An other solution is to not send these graphic elements to the front (better for performance).

Reply:
I don't think that we should use CSS for that, we can simply not tell Sprotty about the hidden elements. We should just ignore those hidden elements in the `convertDiagram` function in the frontend